ociinstaller/plugin.go

```diff
@@ -126,18 +128,59 @@ func installPluginConfigFiles(image *SteampipeImage, tempdir string) error {
 	for _, obj := range objects {
 		sourceFile := filepath.Join(sourcePath, obj.Name())
 		destFile := filepath.Join(installTo, obj.Name())
-
-		if err := copyFileUnlessExists(sourceFile, destFile); err != nil {
-			return fmt.Errorf("could not copy %s to %s", sourceFile, destFile)
-
+		if err := copyConfigFileUnlessExists(sourceFile, destFile, image.ImageRef); err != nil {
+			return fmt.Errorf("could not copy config file from %s to %s", sourceFile, destFile)
 		}
 	}
 
 	return nil
 }
 
-func pluginInstallDir(imageRef string) string {
-	ref := NewSteampipeImageRef(imageRef)
+func copyConfigFileUnlessExists(sourceFile string, destFile string, ref *SteampipeImageRef) error {
+	if fileExists(destFile) {
+		return nil
+	}
+	inputData, err := os.ReadFile(sourceFile)
+	if err != nil {
+		return fmt.Errorf("couldn't open source file: %s", err)
+	}
+	inputStat, err := os.Stat(sourceFile)
+	if err != nil {
+		return fmt.Errorf("couldn't read source file permissions: %s", err)
+	}
+	// transform
+	inputData = addPluginStreamToConfig(inputData, ref, "latest")
+	if err = os.WriteFile(destFile, inputData, inputStat.Mode()); err != nil {
+		return fmt.Errorf("writing to output file failed: %s", err)
+	}
+	return nil
+}
+
+func addPluginStreamToConfig(src []byte, ref *SteampipeImageRef, nopIfStream string) []byte {
+	_, _, stream := ref.GetOrgNameAndStream()
+	if stream == nopIfStream {
+		return src
+	}
+
+	regex := regexp.MustCompile(`^(\s*)plugin\s*=\s*"(.*)"\s*$`)
+	substitution := fmt.Sprintf(`$1 plugin = "$2@%s"`, stream)
+
+	srcScanner := bufio.NewScanner(strings.NewReader(string(src)))
+	srcScanner.Split(bufio.ScanLines)
+	destBuffer := bytes.NewBufferString("")
+
+	for srcScanner.Scan() {
+		line := srcScanner.Text()
+		if regex.MatchString(line) {
+			line = regex.ReplaceAllString(line, substitution)
+			line = line[1:]
```

Comment:
why?

Reply:
This is because in the substitution `regex` I had to put in an extra space to isolate the substitution token `$1`. Removing it here.